content/download-self.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// preventing reregistration if reinjected by tabs.executeScript for whatever reason, just in case
+if (typeof self.oldCode !== 'string') {
+  chrome.runtime.onConnect.addListener(port => {
+    if (port.name !== 'downloadSelf') return;
+    const read = r => r.status === 200 ? r.text() : Promise.reject(r.status);
+    const postBack = msg => {
+      port.postMessage(msg);
+      self.oldCode = msg.code;
+    };
+    port.onMessage.addListener(({id, timer}) => {
+      fetch(location.href, {mode: 'same-origin'})
+        .then(read)
+        .then(code => ({id, code: timer && code === self.oldCode ? null : code}))
+        .catch(error => ({id, error}))
+        .then(postBack);
+    });
+  });
+}
+
+// this assignment also passes the result to tabs.executeScript
+self.oldCode = (document.querySelector('body > pre') || document.body).textContent;
```

Comment:
This resets `oldCode` to the initial state when `oldCode` is already set and the content script is injected multiple times. It probably should be:
```js
self.oldCode = self.oldCode || (...);
```

Reply:
Not really, it'll just cause the correct current code to be sent once again to the port. This construction is also dangerous as it may resolve to an element in DOM with `id="oldCode"` so I would have to use another check or use an IIFE which I really don't want to. Let's keep it simple, just the way it is.

Reply:
Resolved by moving it inside `if`.